plugins/modules/win_scoop.ps1

```diff
@@ -0,0 +1,230 @@
+    if ([Net.SecurityProtocolType].GetMember("Tls11").Count -gt 0) {
+      $security_protocols = $security_protcols -bor [Net.SecurityProtocolType]::Tls11
+    }
+    if ([Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
+      $security_protocols = $security_protcols -bor [Net.SecurityProtocolType]::Tls12
+    }
+    [Net.ServicePointManager]::SecurityProtocol = $security_protocols
+
+    $client = New-Object -TypeName System.Net.WebClient
+
+    $script_url = "https://get.scoop.sh"
+
+    try {
+      $install_script = $client.DownloadString($script_url)
+    }
+    catch {
+      $module.FailJson("Failed to download Scoop script from '$script_url'; $($_.Exception.Message)", $_)
+    }
+
+    if (-not $module.CheckMode) {
+      $res = Run-Command -Command "powershell.exe -" -stdin $install_script -environment $environment
+      if ($res.rc -ne 0) {
+        $module.Result.rc = $res.rc
+        $module.Result.stdout = $res.stdout
+        $module.Result.stderr = $res.stderr
+        $module.FailJson("Scoop bootstrap installation failed.")
+      }
+      $module.Warn("Scoop was missing from this system, so it was installed during this task run.")
+    }
+    $module.Result.changed = $true
+
+    # Refresh PATH
+    $env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path", "User")
+
+    # locate the newly installed scoop.ps1
+    $scoop_app = Get-Command -Name scoop.ps1 -Type ExternalScript -ErrorAction SilentlyContinue
+  }
+
+  if ($module.CheckMode -and $null -eq $scoop_app) {
+    $module.Result.skipped = $true
+    $module.Result.msg = "Skipped check mode run on win_scoop as scoop.ps1 cannot be found on the system"
+    $module.ExitJson()
+  }
+
+  if ($null -eq $scoop_app -or -not (Test-Path -LiteralPath $scoop_app.Path)) {
+    $module.FailJson("Failed to find scoop.ps1, make sure it is added to the PATH")
+  }
+
+  return $scoop_app.Path
+}
+
+function Get-ScoopPackages {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path
+  )
+
+  $command = Argv-ToString -arguments @("powershell.exe", $scoop_path, "export")
+  $res = Run-Command -Command $command
+  if ($res.rc -ne 0) {
+    $module.Result.command = $command
+    $module.Result.rc = $res.rc
+    $module.Result.stdout = $res.stdout
+    $module.Result.stderr = $res.stderr
+    $module.FailJson("Error checking installed packages")
+  }
+
+  $installed_packages = $res.stdout -split "`n" |
+  Select-String '(.*?) \(v:(.*?)\) \[(.*?)\]' |
+  ForEach-Object {
+    New-Object PSObject -Property ([Ordered]@{
+        "Package" = $_.Matches[0].Groups[1].Value
+        "Version" = $_.Matches[0].Groups[2].Value
+        "Bucket"  = $_.Matches[0].Groups[3].Value
+      })
+  }
+
+  return $installed_packages
+}
+
+function Get-InstallScoopPackageArguments {
+  $arguments = [System.Collections.ArrayList]@()
```

Comment:
I know this is probably copied from win_chocolatey but if you use a generic list you don't need to discard the output from .Add()

```powershell
$arguments = [System.Collections.Generic.List[String]]@()

$arguments.Add('--arch')
$arguments.Add($arch)

...
```

In any case this comes back to an earlier point I made in a comment above. Because we are calling `scoop.ps1` we shouldn't have to escape these arguments. We are passing the string as an object using PowerShell's rules and I hope that scoop respects that.